consolidate/mesher/meshes.py

```diff
@@ -7,90 +13,90 @@ def __init__(self, deck,geometry):
         self.deck = deck
         self.geometry=geometry
         self.set_mesh_grid() 
-        self.set_temperatures()
-        self.set_conductivity()
-        self.set_density()
-        self.set_specific_heat()
-        self.set_diffusivity()
-        self.set_viscosity()   
-        self.set_dic()
-        self.set_heat_density()
-        
+        self.set_dx()
+        self.set_dy()
+        self.set_Element_Position()
         
         
     def set_mesh_grid(self):
-        self.nx = int(self.deck.doc["Simulation"]["Number of Elements X"])
-        self.ny = int(self.deck.doc["Simulation"]["Number of Elements Y"])
-        self.dx=self.geometry.Lx/self.nx
-        self.dy=self.geometry.Ly/self.ny
-        X, Y = np.meshgrid(np.arange(0, self.ny), np.arange(0, self.nx))
-        X=X[1,:].copy()
-        Y=Y[:,1].copy()
-        self.X = X
-        self.Y = Y
+        self.nx = int(self.deck.doc["Simulation"]["Number of Elements X"])        
+        self.ny =  {
+  "Top Adherent": int(self.deck.doc["Simulation"]["Number of Elements Top Plate Thickness"]) ,
+  "HE"       : int(self.deck.doc["Simulation"]["Number of Elements HE"]),
+  "Bottom Adherent": int(self.deck.doc["Simulation"]["Number of Elements Bottom Plate Thickness"]) 
+                  }
 
-        
-        self.ny1= int(self.ny/2)
 
+        self.number_of_elements_Y=self.ny.get("Top Adherent")+self.ny.get("HE")+self.ny.get("Bottom Adherent")
```

Comment:
Where do you use self.number_of_elements_Y after?

Reply:
in plotting...

Reply:
Could you please detail a bit more what it is you mean ?